backends/platform/ios7/ios7_micro_gamepad_controller.mm

```diff
@@ -0,0 +1,93 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// Disable symbol overrides so that we can use system headers.
+#define FORBIDDEN_SYMBOL_ALLOW_ALL
+
+#include "common/events.h"
+#include "backends/platform/ios7/ios7_micro_gamepad_controller.h"
+#include "backends/platform/ios7/ios7_video.h"
+#include <GameController/GameController.h>
+
+@implementation MicroGamepadController {
+#ifdef __IPHONE_9_0
+	GCMicroGamepad *_microGamepad;
+#endif
+}
+
+@dynamic view;
+@dynamic isConnected;
+
+- (id)initWithView:(iPhoneView *)view {
+	self = [super initWithView:view];
+
+	if (self) {
+		[[NSNotificationCenter defaultCenter] addObserver:self
+												 selector:@selector(microGamepadDidConnect:)
+													 name:@"GCControllerDidConnectNotification"
+												   object:nil];
+	}
+
+	return self;
+}
+
+- (void)microGamepadDidConnect:(NSNotification *)notification {
+#ifdef __TVOS_9_0
+	[self setIsConnected:YES];
+	
+	GCController* gccontroller = notification.object;
+	if (!gccontroller.microGamepad) {
+		NSLog(@"GCMicroGamepad not supported by that gamepad!");
+		return;
+	}
+	
+	_microGamepad = gccontroller.microGamepad;
+	
+	_microGamepad.dpad.valueChangedHandler = ^(GCControllerDirectionPad * _Nonnull dpad, float xValue, float yValue) {
+		// The touchpad on the remote can be used as a D-pad.
+		// The D-pad provides analog input data which makes it logical to
+		// treat it as a joystick.
+		// Convert the given axis values in float (-1 to 1) to ScummVM Joystick
+		// Axis value as integers (0 to int16_max)
+		int x = xValue * (float)Common::JOYAXIS_MAX;
+		int y = yValue * (float)Common::JOYAXIS_MAX;
+
+		// Apple's Y values are reversed from ScummVM's
+		[self handleJoystickAxisMotionX:x andY:0-y forJoystick:kGameControllerJoystickLeft];
+	};
+	
+	_microGamepad.buttonA.valueChangedHandler = ^(GCControllerButtonInput * _Nonnull button, float value, BOOL pressed) {
+		[self handleMouseButtonAction:kGameControllerMouseButtonLeft isPressed:pressed at:[[self view] pointerPosition]];;
+	};
+	
+	_microGamepad.buttonX.valueChangedHandler = ^(GCControllerButtonInput * _Nonnull button, float value, BOOL pressed) {
+		[self handleMouseButtonAction:kGameControllerMouseButtonRight isPressed:pressed at:[[self view] pointerPosition]];
+	};
+
+#ifdef __TVOS_13_0
+	_microGamepad.buttonMenu.valueChangedHandler = ^(GCControllerButtonInput * _Nonnull button, float value, BOOL pressed) {
+		[[self view] addEvent:InternalEvent(kInputMainMenu, 0, 0)];
```

Comment:
Should check pressed value as well. Else it will generate two events, one when pressed then one when released